### PR TITLE
Version management, platform validation and fallback colors

### PR DESCRIPTION
**Pull request by felipecastrosales**
This PR introduces version management, platform validation, and several improvements to the STAC framework.

### Version Management System
- **StacVersion Class**: New utility for managing app version constraints with build numbers
  - Supports platform-specific version configurations (iOS/Android)
  - Provides condition-based version checks (greaterThan, lessThan, equal, etc.)
  - Integrates with widget rendering to conditionally display components based on app version

### Build Number Support
- Added `buildNumber` parameter to `Stac.initialize()`
- Integrated build number registration in `StacRegistry`
- Enables version-aware widget rendering from JSON

### Platform Validation
- Widgets can now specify supported platforms
- Automatic platform checks with fallback behavior for unsupported platforms
- Warning logs for platform mismatches

### Custom Color Parsing
- Added `parseCustomColor` hook in `StacRegistry`
- Allows custom color resolution for non-standard color identifiers
- Accessible via `StacService.setParseCustomColor()`

## 🔧 Improvements

- **New Layout Parsers**: Added centralized export for foundation layout parsers
- **Widget Exports**: Exposed `StacDouble` and `StacText` parsers
- **Dependency Update**: Switched `stac_core` to local path dependency for development

## ✅ Tests

- Comprehensive test suite for `StacVersion` (277 lines)
  - Platform-specific configuration tests
  - Version condition parsing and validation
  - Build number satisfaction logic
  - Edge case handling

## 🧹 Chores

- Standardized quote styles across pubspec.yaml files (single → double quotes)
- Minor formatting improvements and cleanup

## 🔄 Migration Notes

If you're using `Stac.initialize()`, you can now optionally pass a `buildNumber`:

```dart
await Stac.initialize(
  json: json,
  buildNumber: 123, // Optional: enables version-aware widgets
);
```

To use custom color parsing:

```dart
StacService.setParseCustomColor((colorString) {
  // Your custom color parsing logic
  return Colors.purple;
});
```

## Summary by CodeRabbit

* **New Features**
  * Build-number support in initialization and content gating, plus a new StacVersion utility for build/condition rules.
  * Platform-aware gating and platform detection utilities for selective content.
  * Custom color parsing hook and a public setter to register it.
  * New public exports for layout and widget parsers; unified copyWith extension for text styles.

* **Tests**
  * Comprehensive tests covering version/build gating, platform handling, and condition parsing.

### CODE REVIEW

**Comment by felipecastrosales** [comment]
@coderabbitai enhance my description on pr.

File: packages/stac/lib/src/framework/stac_registry.dart [hunk 1]
```diff
@@ -68,6 +72,12 @@ class StacRegistry {
     });
   }
 
+  void registerBuildNumber(int? buildNumber) {
+    if (buildNumber != null) {
+      _buildNumber = buildNumber;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac/lib/src/framework/stac_registry.dart:79]: `reset` is not necessary, cuz `buildNumber` cannot change in runtime. 

this will be setted in initialization of the app

File: packages/stac/lib/stac.dart [hunk 2]
```diff
@@ -4,6 +4,8 @@ export 'package:stac/src/parsers/actions/actions.dart';
 export 'package:stac/src/parsers/parsers.dart';
 export 'package:stac/src/services/services.dart';
 export 'package:stac/src/utils/utils.dart';
+export 'package:stac/src/parsers/foundation/layout/parsers.dart';
```
[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac/lib/stac.dart]: ok

File: packages/stac/lib/src/framework/stac_service.dart [hunk 3]
```diff
@@ -190,18 +188,65 @@ class StacService {
...
+
+        // If platform is not supported, log and return null
+        if (!isPlatformSupported) {
+          final platformsStr = supportedPlatforms.join(', ');
+          Log.w(
+            'Widget not supported on platform [$currentPlatform]. Only available for: $platformsStr',
+          );
+          return null;
+        }
+      }
```
[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac/lib/src/framework/stac_service.dart:250]: fixed or not? now

[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac/lib/src/framework/stac_service.dart:250]: fixed or not? now @coderabbitai

File: packages/stac/lib/src/utils/version/stac_version.dart [hunk 4]
```diff
@@ -0,0 +1,281 @@
+import 'dart:io';
```
[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac/lib/src/utils/version/stac_version.dart]: fixed now? @coderabbitai

File: packages/stac/lib/src/utils/version/stac_version.dart [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+  /// Converts a string operator to its corresponding [StacConditionVersion].
+  StacConditionVersion toStacConditionVersion() => switch (this) {
+        '>' => StacConditionVersion.greaterThan,
+        '>=' => StacConditionVersion.greaterThanOrEqual,
+        '<' => StacConditionVersion.lessThan,
+        '<=' => StacConditionVersion.lessThanOrEqual,
+        '==' => StacConditionVersion.equal,
+        '!=' => StacConditionVersion.notEqual,
+        _ => StacConditionVersion.notEqual,
+      };
```
[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac/lib/src/utils/version/stac_version.dart]: no, default is `notEqual` to ensure behaviour would be expected values.

File: packages/stac/test/src/utils/version/stac_version_test.dart [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+      test('handles version with non-numeric components', () {
+        StacRegistry.instance.registerBuildNumber(2000);
+
+        final version = StacVersion(
+          buildNumber: 2000,
+          condition: StacConditionVersion.equal,
+        );
+        // Non-numeric components should be treated as 0
+        expect(version.isSatisfied(StacRegistry.instance.buildNumber), true);
+      });
```
[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac/test/src/utils/version/stac_version_test.dart]: fixed? @coderabbitai

File: packages/stac/test/src/utils/version/stac_version_test.dart [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+        'buildNumber_ios': 2500, // Different platform
+        'condition': '>=',
+      };
+
+      final version = StacVersion.fromJson(json);
+
+      // Should use generic buildNumber since `platform`-specific is not available
+      expect(version.buildNumber, 1500);
+      expect(version.condition, StacConditionVersion.greaterThanOrEqual);
+    });
```
[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac/test/src/utils/version/stac_version_test.dart:204]: fixed? @coderabbitai

File: packages/stac/test/src/utils/version/stac_version_test.dart [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+      });
+
+      test('returns false when app version is null', () {
+        StacRegistry.instance.registerBuildNumber(null);
+        final version = StacVersion(
+          buildNumber: 1000,
+          condition: StacConditionVersion.equal,
+        );
+        expect(version.isSatisfied(StacRegistry.instance.buildNumber), false);
+      });
```
[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac/test/src/utils/version/stac_version_test.dart]: fixed? @coderabbitai

File: packages/stac_webview/pubspec.yaml [hunk 9]
```diff
@@ -19,7 +19,12 @@ dependencies:
     sdk: flutter
   webview_flutter: ^4.13.1
   json_annotation: ^4.10.0
-  stac_core: ^1.3.0
+  # when approved, use stac_core: ^1.3.x
+  stac_core:
+    git:
+      url: https://github.com/SuaMusica/stac.git
+      path: packages/stac_core
+      ref: devall
```
[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac_webview/pubspec.yaml:27]: no needed

File: packages/stac/pubspec.yaml [hunk 10]
```diff
@@ -23,7 +23,12 @@ dependencies:
   cached_network_image: ^3.4.1
   flutter_svg: ^2.2.3
   stac_logger: ^1.1.0
-  stac_core: ^1.3.0
+  # when approved, use stac_core: ^1.3.x
+  stac_core:
+    git:
+      url: https://github.com/SuaMusica/stac.git
+      path: packages/stac_core
+      ref: devall
```
[the remark this answers is not in the document]
Comment (felipecastrosales) [reply packages/stac/pubspec.yaml:31]: no needed

**Comment by felipecastrosales** [comment]
hey @divyanshub024 can u review this?
